### PR TITLE
Refactor applications

### LINKED ISSUE #312: Separate applicant status from user status
Also don't give collegist role until they are admitted. This would simplify things.

### PR DESCRIPTION
**Pull request by kdmnk**
## Description

This is manly just to split the application and application handling part of previous ApplicationController into a new AdmissionController to prepare for #332

Other changes:
- removed statuses, instead now we only have "submitted" boolean
- detached applications from Collegist roles, now it has 'applied_for_resident_status' boolean
- found some bugs mostly in the UI
- commented out finalise functions, will fix it later
- refactored some code for better readability

More changes will come in a later PR.

### CODE REVIEW

File: resources/views/auth/admission/index.blade.php [hunk 1]
```diff
@@ -5,36 +5,32 @@
...
-                                        style="padding-left: 25px; margin: 5px">@include('auth.application.status', ['status' => $st])</span>
-                                </label>
-                            @endforeach
-                        @endcan
+                        @can('viewUnfinished', \App\Models\Application::class)
+                            <label>
+                                <input type="checkbox" name="submitted" checked>
+                                <span style="padding-left: 25px; margin: 5px">Véglegesített</span>
+                            </label>
+                        @endif
```
Comment (viktorcsimma) [inline resources/views/auth/admission/index.blade.php:21]: This does not seem to work for me (it does not update anything and is always checked after reload); can you check it out? Or am I missing something?

File: resources/views/auth/admission/index.blade.php [hunk 2]
```diff
@@ -54,13 +50,13 @@
...
             @include('auth.application.application', ['user' => $application->user, 'expanded' => false])
         </a>
     @endforeach
     <hr>
     <h6>Összesen: <b class="right">{{$applications->count()}} jelentkező</b></h6>
-    @can('finalize', \App\Models\ApplicationForm::class)
+    @can('finalize', \App\Models\Application::class)
         @if($applicationDeadline?->addWeeks(1) < now())
             <div class="card" style="margin-top:20px">
                 <div class="card-content">
```
Comment (viktorcsimma) [inline resources/views/auth/admission/index.blade.php:62]: Here, the comments about the "Submitted" and "Called in" statuses are not up-to-date.

File: app/Http/Controllers/Auth/AdmissionController.php [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+        }
+        return redirect()->back();
+    }
+
+    /**
+     * Accept and delete applciations.
+     * @return RedirectResponse
+     * @throws AuthorizationException
+     */
+    public function finalize(): RedirectResponse
```
Comment (viktorcsimma) [inline app/Http/Controllers/Auth/AdmissionController.php:141]: How do we plan to do this? I don't know how we should differentiate between admitted and banished applicants if there are no statuses anymore.

File: routes/web.php [hunk 4]
```diff
@@ -171,11 +169,15 @@
...
-    Route::post('/applications', [ApplicationController::class, 'editApplication'])->name('applications.edit');
-    Route::get('/applications/export', [ApplicationController::class, 'exportApplications'])->name('applications.export');
+    /** Admission and applicants handling */
+    Route::post('/admission/period/update', [AdmissionController::class, 'updateApplicationPeriod'])->name('admission.period.update');
+    Route::get('/admission/export', [AdmissionController::class, 'export'])->name('admission.export');
+    Route::post('/admission/finalize', [AdmissionController::class, 'finalize'])->name('admission.finalize');
+
+    Route::get('/applicants', [AdmissionController::class, 'index'])->name('admission.applicants.index');
+    Route::get('/applicants/{application}', [AdmissionController::class, 'show'])->name('admission.applicants.show');
+    Route::get('/applicants/{application}/edit', [AdmissionController::class, 'edit'])->name('admission.applicants.edit');
```
Comment (viktorcsimma) [inline routes/web.php]: This seems to be a POST request for me.

File: app/Http/Controllers/Auth/ApplicationController.php [hunk 5]
```diff
@@ -97,256 +72,60 @@ public function showApplicationForm(Request $request): View|RedirectResponse
...
-        }
-    }
-
-    /**
-     * Edit an application's note.
-     * @param Request $request
-     * @return RedirectResponse
-     * @throws AuthorizationException
-     */
-    public function editApplication(Request $request): RedirectResponse
```
Comment (viktorcsimma) [inline app/Http/Controllers/Auth/ApplicationController.php:192]: This seems to be missing from AdmissionController.

File: resources/views/auth/application/application.blade.php [hunk 6]
```diff
@@ -11,15 +11,9 @@
...
-                            <span class="right">
-                                @include('auth.application.status', ['status' => $user->application->status])
-                            </span>
-                        @else
-                            @livewire('application-status-update', ['application' => $user->application])
-                        @endif
-                    @endcan
+{{--                    @can('editStatus', \App\Models\Application::class)--}}
+{{--                        @livewire('application-status-update', ['application' => $user->application])--}}
+{{--                    @endcan--}}
```
Comment (viktorcsimma) [inline resources/views/auth/application/application.blade.php:16]: Here, we should somehow indicate whether the application has been submitted.

**Review by viktorcsimma — CHANGES_REQUESTED** [review]
Seems to have some nice ideas:) but we probably cannot merge it until the deadline in a stable state. My fault; I could have noticed it earlier.

My main objection: should we really abolish statuses for a boolean flag? It was useful to see at one place whether they have been called in, banished or admitted. How would we solve finalization without these?

But until then, I'll integrate the changes to the instructions into #572; I like them:)

File: resources/views/auth/application/files.blade.php [hunk 7]
```diff
@@ -29,49 +25,35 @@
...
-              enctype='multipart/form-data'>
-            @csrf
-            <div class="card-content">
-                <div class="card-title">Feltöltés</div>
+            <form method="POST" action="{{ route('application.store', ['page' => 'files']) }}"
+                  enctype='multipart/form-data'>
+                @csrf
                 <div class="row">
-                    <x-input.file s=12 m=6 id="file" size="2000000" accept=".pdf,.jpg,.png,.jpeg" text="general.browse" required/>
+                    <x-input.file s=12 m=6 id="file" size="2000000" accept=".pdf,.jpg,.png,.jpeg" text="general.browse" helper=".pdf,.jpg,.png,.jpeg fájlok tölthetőek fel, maximum 2 MB-os méretig." required/>
```
Comment (viktorcsimma) [inline resources/views/auth/application/files.blade.php:32]: The helper does not seem to work for me.

File: resources/views/utils/user/profile-picture.blade.php [hunk 8]
```diff
@@ -24,28 +24,28 @@
             @csrf
             <div class="card-action valign-center">
                 <x-input.file s="12" l="7" xl="8" id="picture" style="margin-top:auto" accept=".jpg,.png,.jpeg"
-                                text="general.browse" required/>
+                                text="general.browse" required helper=".jpg,.png,.jpeg fájlok tölthetőek fel, maximum 2 MB-os méretig."/>
```
Comment (viktorcsimma) [inline resources/views/utils/user/profile-picture.blade.php:27]: Same here.

File: app/Http/Controllers/Auth/AdmissionController.php (hunk 3 above)
Comment (kdmnk) [reply app/Http/Controllers/Auth/AdmissionController.php:141]: that will come later in a future PR. We have the statuses in https://github.com/EotvosCollegium/mars/pull/574

File: resources/views/auth/admission/index.blade.php (hunk 2 above)
Comment (kdmnk) [reply resources/views/auth/admission/index.blade.php:62]: See https://github.com/EotvosCollegium/mars/pull/574

File: resources/views/auth/application/application.blade.php (hunk 6 above)
Comment (kdmnk) [reply resources/views/auth/application/application.blade.php:16]: See https://github.com/EotvosCollegium/mars/pull/574

File: resources/views/auth/application/files.blade.php (hunk 7 above)
Comment (kdmnk) [reply resources/views/auth/application/files.blade.php:32]: I updated the file component. you need to reset cache.

File: resources/views/utils/user/profile-picture.blade.php (hunk 8 above)
Comment (kdmnk) [reply resources/views/utils/user/profile-picture.blade.php:27]: I updated the file component. you need to reset cache.

File: resources/views/auth/admission/index.blade.php (hunk 1 above)
Comment (kdmnk) [reply resources/views/auth/admission/index.blade.php:21]: Looks like I fixed this in #574 so you can just ignore

File: resources/views/auth/admission/index.blade.php (hunk 2 above)
Comment (kdmnk) [reply resources/views/auth/admission/index.blade.php:62]: No, actually, you are right, I see what you mean now. I will fix this quickly.

File: resources/views/auth/application/app.blade.php [hunk 9]
```diff
@@ -49,34 +53,35 @@
...
     @endif
-    @if($user->application->status != App\Models\ApplicationForm::STATUS_IN_PROGRESS && isset($user->internetAccess))
+    @if($user->application->submitted)
         @include('network.internet.wifi_password', ['internet_access' => $user->internetAccess, 'application' => true])
     @endif
 
+    @if(request()->get('page') != 'submit')
+    <x-input.button href="{{ route('application', ['page' => 'submit']) }}"
+       class="right coli blue" text="Ellenőrzés és véglegesítés" />
+    @endif
```
Comment (viktorcsimma) [inline resources/views/auth/application/app.blade.php]: For me, the cookie disclaimer covers the button. I think we should simply left the submission option where it originally was; I would look for it there.

**Review by viktorcsimma — CHANGES_REQUESTED** [review]
One more thing; then I think we should merge #574 and #572 here. It's better to test them together.

File: resources/views/auth/application/app.blade.php (hunk 9 above)
Comment (kdmnk) [reply resources/views/auth/application/app.blade.php]: That was also hidden on some special cases (on mobiles). It was not clear that the navbar is scrollable.

Comment (kdmnk) [reply resources/views/auth/application/app.blade.php]: Do you think the cookie disclaimer would be an issue? I can add more padding.

Comment (viktorcsimma) [reply resources/views/auth/application/app.blade.php]: I see… we can move it to a button then; just let’s write some hint on where it is. E.g. _”Miután minden szükséges mezőt megválaszolt és fájlt feltöltött, kérjük, véglegesítse a jelentkezését **a lap alján levő gombra kattintva**.”_

And yes, please add some more padding.

File: routes/web.php (hunk 4 above)
Comment (kdmnk) [reply routes/web.php]: That function does not even exist, I will delete the route.

File: app/Http/Controllers/Auth/ApplicationController.php (hunk 5 above)
Comment (kdmnk) [reply app/Http/Controllers/Auth/ApplicationController.php:192]: That is update in the AdmissionController

**Pull request merged**